### PR TITLE
Feature/em 798 claim slp from axie origin

### PR DESCRIPTION
**Pull request by yfleung3**

### CODE REVIEW

**Review by Disturbing — APPROVED** [review]
LGTM - Make sure to inform to tiff that we do not support free claiming transactions at the moment.

This can be a feature in the backlog.

File: pkg/client/client.go [hunk 1]
```diff
@@ -196,12 +211,13 @@ func (c *AxieClient) createTransactOps(ctx context.Context, privateKeyStr string
 	if err != nil {
 		return nil, err
 	}
+
 	rand.Seed(time.Now().UnixNano())
-	gasPrice := rand.Intn(198888-165313+1) + 165313
+	gasLimit := uint64(rand.Int63n(upperGasLimit-lowerGasLimit+1) + lowerGasLimit)
```
Comment (hcho1989) [inline pkg/client/client.go:216]: why random gasLimit

File: pkg/client/client.go [hunk 2]
```diff
@@ -425,6 +441,174 @@ func (c *AxieClient) GetClaimPayload(ctx context.Context, address, privateKeyStr
...
+	return &resp, nil
+}
+
+func (c *AxieClient) getWithdrawSlpPayload(token string, amount int) (*OriginWithdrawItemsResponse, error) {
+	var headers = utils.DefaultAxieSiteRequestHeaders
+	headers["Authorization"] = fmt.Sprintf("Bearer %s", token)
+	var resp OriginWithdrawItemsResponse
+
+	respBytes, err := utils.CallPostHttpApi(
+		"https://game-api-origin.skymavis.com/v2/rpc/items/withdraw", map[string][]map[string]interface{}{
```
Comment (hcho1989) [inline pkg/client/client.go:516]: shall the url goes to constants?

File: pkg/client/client.go [hunk 3]
```diff
@@ -425,6 +441,174 @@ func (c *AxieClient) GetClaimPayload(ctx context.Context, address, privateKeyStr
...
+	}
+	return c.getClaimableSlpPayload(token)
+}
+
+func (c *AxieClient) getClaimableSlpPayload(token string) (*OriginClaimableResponse, error) {
+	var headers = utils.DefaultAxieSiteRequestHeaders
+	headers["Authorization"] = fmt.Sprintf("Bearer %s", token)
+	var resp OriginClaimableResponse
+	respBytes, err := utils.CallGetHttpApi(
+		"https://game-api-origin.skymavis.com/v2/users/me/items/marketplace/slp",
```
Comment (hcho1989) [inline pkg/client/client.go:495]: shall the url goes to constants?

**Review by hcho1989 — APPROVED** [review]
in general looks good, but consider moving the url to constants

**Pull request merged**